malcolm/modules/ADCore/parts/detectordriverpart.py

```diff
@@ -299,29 +297,38 @@ def on_configure(
         context.unsubscribe_all()
         child = context.block_view(self.mri)
         self.check_driver_version(child)
-        self.setup_detector(
-            context,
-            completed_steps,
-            steps_to_do,
-            generator.duration,
-            part_info,
-            **kwargs,
-        )
+
         # Calculate how long to wait before marking this scan as stalled
         self.frame_timeout = FRAME_TIMEOUT
         if generator.duration > 0:
             self.frame_timeout += generator.duration
         else:
             # Double it to be safe
             self.frame_timeout += FRAME_TIMEOUT
+
         # If detector can be soft triggered, then we might need to defer
         # starting it until run. Check triggerMode to find out
         if self.soft_trigger_modes:
             mode = child.triggerMode.value
             self.is_hardware_triggered = mode not in self.soft_trigger_modes
+
+        # If detector is hardware triggered we can configure the detector for all frames
+        # now, rather than configuring and arming for each inner scan, and send the
+        # triggers for one inner scan in each run
+        steps_in_run = steps_to_do
         if self.is_hardware_triggered:
-            # Start now if we are hardware triggered
-            self.arm_detector(context)
+            steps_to_do = generator.size
```

Comment:
```suggestion
            steps_to_do = generator.size - completed_steps
```

Reply:
we decided to call it ```steps_until_done```

Reply:
Instead of renaming this I have decided to take out the new steps_in_run parameter entirely and maintain this in steps_to_do, because that is what it represents in on_configure. What I really wanted here is to be able to set numImages to a value other than steps_to_do, i.e. the whole scan rather than the inner scan, so I have just added a num_images parameter instead. I think this is much clearer. We could consider changing steps_to_do to steps_in_run, though.